packages/fuels/src/cli/types.ts

```diff
@@ -0,0 +1,115 @@
+import type { DeployContractOptions } from '@fuel-ts/contract';
+
+export enum Commands {
+  build = 'build',
+  deploy = 'deploy',
+  dev = 'dev',
+  init = 'init',
+}
+
+export type CommandEvent =
+  | {
+      type: Commands.build;
+      data: unknown;
+    }
+  | {
+      type: Commands.deploy;
+      data: DeployedContract[];
+    }
+  | {
+      type: Commands.dev;
+      data: unknown;
+    }
+  | {
+      type: Commands.init;
+      data: unknown;
+    };
+
+export type DeployedContract = {
+  name: string;
+  contractId: string;
+};
+
+export type ContractDeployOptions = {
+  contracts: DeployedContract[];
+  contractName: string;
+  contractPath: string;
+};
+
+export type OptionsFunction = (
+  options: ContractDeployOptions
+) => DeployContractOptions | Promise<DeployContractOptions>;
+
+export type UserFuelsConfig = {
+  /** Relative directory path to Forc workspace */
+  workspace?: string;
+  /** Relative directory path to Sway contracts */
+  contracts?: string[];
+  /** List of relative directory paths to Sway predicates */
+  predicates?: string[];
+  /** List of relative directory paths to Sway scripts */
+  scripts?: string[];
+  /** Relative directory path for generating Typescript definitions */
+  output: string;
+
+  /**
+   * Wallet private key, used when deploying contracts.
+   * Should ideally come from env — `process.env.MY_PRIVATE_KEY`
+   */
+  privateKey?: string;
+
+  /**
+   * Contracts will be deployed using this provider.
+   * Default: http://localhost:4000
+   */
+  providerUrl?: string;
+
+  /** Relative filepath to custom `chainConfig.json` to be used when initializing a Fuel Core node */
+  chainConfig?: string;
+
+  /** Static of dynbamic deploy configs to be used when deploying contracts */
+  deployConfig?: DeployContractOptions | OptionsFunction;
+
+  /** Opt-in or out from using builtin Forc binaries */
+  useBuiltinForc?: boolean;
```

Comment:
The "built-in" wording initially confused me; I had questions like, "Built into what?" It seems to me that something like `usePackagedForc` and `usePackagedFuelCore` would be more descriptive of the nature of the binaries. Perhaps even something like this:

```ts
{
    /** Select which forc binary to use. Defaults to `system`. If forc isn't found on the system, the CLI warns the user and uses the packaged binary instead. */
    forcBinary: 'system' | 'packaged';
    /** Select which fuel-core binary to use. Defaults to `system`. If fuel-core isn't found on the system, the CLI warns the user and uses the packaged binary instead. */
    fuelCoreBinary: 'system' | 'packaged';
}
```

The discriminated union spells out all the possibilities, whereas `useBuiltinForc`/`useBuiltinFuelCore` implies that the user knows that the system `forc`/`fuel-core` can be used and is used by default.

Reply:
Thanks for bringing my attention to this confusion.

This was a tricky choice, yet I think `built-in` fits better.

I reviewed and improved the docs to ensure proper disambiguation.

https://github.com/FuelLabs/fuels-ts/pull/1291/commits/a1a39dd3959c3010f78b466a0a554c187c847497